drivers/adc/adc_ltc2451.c

```diff
@@ -0,0 +1,102 @@
+/* LLTC LTC2451 ADC
+ *
+ * Copyright (c) 2023 Brill Power Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(ltc2451, CONFIG_ADC_LTC2451_LOG_LEVEL);
+
+#define DT_DRV_COMPAT lltc_ltc2451
+
+struct ltc2451_config {
+	struct i2c_dt_spec i2c;
+};
+
+struct ltc2451_data {
+	uint8_t conversion_speed;
+};
+
+static int ltc2451_channel_setup(const struct device *dev,
+				const struct adc_channel_cfg *channel_cfg)
+{
+	(void)dev;
+	(void)channel_cfg;
+	return -ENOSYS; /** there is only 1 channel which requires no real setup */
+}
+
+static int ltc2451_set_conversion_speed(const struct device *dev, uint8_t conversion_speed)
+{
+	const struct ltc2451_config *config = dev->config;
+	struct ltc2451_data *data = dev->data;
+	uint8_t wr_buf[1] = {conversion_speed};
+
+	int err = i2c_write(config->i2c.bus, wr_buf, sizeof(wr_buf), config->i2c.addr);
```

Comment:
Use the dt variants (`i2c_write_dt` and friends) for all I2C transactions

Reply:
Has been changed